### PR TITLE
fortran sizeof: ensure mpi_sizeof*f90 is not empty

### PR DESCRIPTION
**Pull request by jsquyres**
Per http://www.open-mpi.org/community/lists/devel/2015/08/17775.php, some compilers don't like it when there's a .f90 file that only contains comments (and no actual Fortran code).  So if OMPI determines that the Fortran compiler does not support enough Fortran mojo to support MPI_SIZEOF, generate at least one dummy Fortran subroutine that can be compiled in an otherwise barren Fortran landscape that is devoid of life and hope.

(cherry picked from commit open-mpi/ompi@d5763a8288c994e1d55a333d45f1a85d64341aff)

Thanks to @PHHargrove for raising the issue.

### CODE REVIEW

File: ompi/mpi/fortran/base/gen-mpi-sizeof.pl [hunk 1]
```diff
@@ -218,12 +218,23 @@ sub output_file {
...
-! compiler to build Open MPI.\n\n";
+! compiler to build Open MPI.
+
+! Dummy subroutine, just so that there is *some* Fortran in this file
+! (some compilers are unhappy if there are no Fortran statements in this
+! file).
+subroutine ompi_sad_panda()
+  implicit none
+
+  print *, 'Open MPI is a sad panda because your Fortran compiler'
```
Comment (hppritcha) [inline ompi/mpi/fortran/base/gen-mpi-sizeof.pl:243]: image

**Comment by hppritcha** [comment]
:+1:
@jsquyres go ahead and merge if the jenkins are happy.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by rhc54** [comment]
@jsquyres VERY sad panda...

**Comment by jsquyres** [comment]
Mmmph.  Somehow the "x" bit got lost on the script.  Fixed -- jenkins will retest shortly...

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
@jsquyres please merge

**Comment by PHHargrove** [comment]
:-1:

As noted in the parallel PR529 this does not fix the reported problem and we don't yet have direct evidence that the empty file was a contributing cause (since the current libtool command doesn't try to link it).

**Comment by jsquyres** [comment]
Linking to #529 -- followups are occurring there...

**Comment by PHHargrove** [comment]
@jsquyres 
Now that you have passed my review on #529, let me know when there is a tarball to test #530.

**Comment by jsquyres** [comment]
@PHHargrove 2.x tarball with this PR applied is available here: http://www.open-mpi.org/~jsquyres/for-paul/

**Comment by PHHargrove** [comment]
@jsquyres 
Test are underway w/ the new tarball.
However, I am unsure when I will have a chance to check results.

**Comment by PHHargrove** [comment]
@jsquyres @hppritcha 
All looks good in my testing on Solaris-11.2 with all 4 combinations of {m32,m64} X {Studio,Gnu}

**Comment by jsquyres** [comment]
Thanks @PHHargrove!

**Pull request merged**